add autofocus info

mrtnmgs opened the pull request:
### Description

When another element than the first focusable element should receive focus, it should be given the `autofocus` attribute.
see https://github.com/WICG/webcomponents/issues/977

### Motivation

Hmm. Make MDN docs better I guess?

### Additional details

It currently doesn't work in Chrome, but I didn't think this info belonged on MDN docs. https://bugs.chromium.org/p/chromium/issues/detail?id=1399465

mrtnmgs committed:
add autofocus info

https://github.com/WICG/webcomponents/issues/977

note: it currently doesn't work in Chrome, but I don't think this info belongs on MDN docs?
https://bugs.chromium.org/p/chromium/issues/detail?id=1399465

estelle commented:
DRIVE BY COMMENT: 
Automatically focusing an element can confuse people using screen readers or with cognitive differences, and can cause the page to scroll, which impacts mobile device users who will also not see preceding content. I am not sure we should ever be recommending autofocus. Is this worth an exception?

see: 
* https://dequeuniversity.com/rules/axe-linter/1.0/no-autofocus
* https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autofocus#accessibility_considerations

sideshowbarker commented:
cc @scottaohara and @stevefaulkner for insight

> DRIVE BY COMMENT: Automatically focusing an element can confuse people using screen readers or with cognitive differences, and can cause the page to scroll, which impacts mobile device users who will also not see preceding content. I am not sure we should ever be recommending autofocus. Is this worth an exception?
> 
> see:
> 
> * [dequeuniversity.com/rules/axe-linter/1.0/no-autofocus](https://dequeuniversity.com/rules/axe-linter/1.0/no-autofocus)
> 
> * [developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autofocus#accessibility_considerations](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autofocus#accessibility_considerations)

scottaohara commented:
In this context, it seems like it should be mentioned.  It is a feature and there while there are footguns to using autofocus, it also does have it's place and can actually be quite important for accessibility if used correctly.

I don't think it makes sense to not mention it, but do think it would be good to at least warn about the potential UX/a11y drawbacks

mrtnmgs commented:
Let me know if you want me to edit this PR to add an explanation / warning.

scottaohara commented:
Yes, @mrtnmgs, that's exactly the sort of nuance I was alluding to but would not necessarily be understood by those not familiar with these sorts of situations. `autofocus` _is meant_ to be used in instances like this, or the dialog element or the proposed popover attributes where we WANT people to use autofocus when necessary for accessibility purposes.

so yeh, if you wouldn't mind taking a stab at adding a note regarding why one would use this for a11y purposes, that'd be wonderful.

mrtnmgs committed:
Added a11y consideration

Added a11y explanation as discussed, as well as a precision concerning the effect of delegatesFocus on programmatic focus.

mrtnmgs commented:
Apologies but I had issues with github's GUI and committed a draft instead of final copy. I'm not seeing a way to fix so will reopen a new PR.
